src/libraries/System.Runtime/tests/System/Text/Unicode/Utf8Tests.cs

```diff
@@ -780,5 +780,148 @@ private static void ToChars_Test_Core(ReadOnlySpan<byte> utf8Input, int destinat
                 Assert.Equal(expectedUtf16Transcoding.ToString(), boundedDestination.Span.Slice(0, actualNumCharsWritten).ToString());
             }
         }
+
```

Comment:
These tests are duplicative of the existing UTF-8 validity tests, and they don't exercise all the same code paths that the existing tests do. They also risk becoming out of sync with the existing tests.

I recommend not adding any new tests. Instead, add the `IsValid` assertion before line 364 here:

https://github.com/dotnet/runtime/blob/f6230892a0a0678c8e78163d13e4ea231584044c/src/libraries/System.Runtime/tests/System/Text/Unicode/Utf8UtilityTests.ValidateBytes.cs#L364

That way the `IsValid` code path will be exercised by the hundreds of tests which already exercise the `GetIndexOfFirstInvalidUtf8Sequence` code paths.